CONTRIBUTING.md

```diff
@@ -1,150 +1,249 @@
+
+### Before Submitting an Enhancement
+
+- Make sure that you are using the latest version.
+- Read the [documentation](https://docs.haystack.deepset.ai/v2.0/docs/intro) carefully and find out if the functionality
+  is already covered, maybe by an individual configuration.
+- Perform a [search](/issues) to see if the enhancement has already been suggested. If it has, add a comment to the
+  existing issue instead of opening a new one.
+- Find out whether your idea fits with the scope and aims of the project. It's up to you to make a strong case to
+  convince the project's developers of the merits of this feature. Keep in mind that we want features that will be
+  useful to the majority of our users and not just a small subset. If you're just targeting a minority of users,
+  consider writing and distributing the integration on your own.
+
+### How Do I Submit a Good Enhancement Suggestion?
+
+Enhancement suggestions are tracked as GitHub issues of type [Feature request for existing integrations](/issues/new?assignees=&labels=feature+request&projects=&template=feature-request-for-existing-integrations.md&title=).
+
+- Use a **clear and descriptive title** for the issue to identify the suggestion.
+- Fill the issue following the template
+
+## Contribute code
+
+> [!IMPORTANT]
+> When contributing to this project, you must agree that you have authored 100% of the content, that you have the
+> necessary rights to the content and that the content you contribute may be provided under the project license.
 
-1. Check if there is already [a related issue](https://github.com/deepset-ai/haystack/issues).
-2. If not, open a new issue to start a quick discussion. Some features might be a nice idea, but don't fit in the scope of Haystack and we hate to close finished PRs!
-3. Once you're ready to start, setup you development environment (see below).
-4. Once you have commits to publish, create a draft pull request with the initial sketch of the implementation and ask for feedback. **Do not wait until the feature is complete!** If this is your first pull request and you wonder how to actually create a pull request, checkout [this manual](https://opensource.com/article/19/7/create-pull-request-github).
-5. Verify that all tests in the CI pass (and add new ones if you implement anything new).
+### Where to start
 
+If this is your first contribution, a good starting point is looking for an open issue that's marked with the label
+["good first issue"](https://github.com/deepset-ai/haystack/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+The core contributors periodically mark certain issues as good for first-time contributors. Those issues are usually
+limited in scope, easy fixable and low priority, so there is absolutely no reason why you should not try fixing them,
+it's a good excuse to start looking into the project and a safe space for experimenting failure: if you don't get the
+grasp of something, pick another one!
```

Comment:
should we also mention that we collect the full list of contributions wanted [here](https://github.com/orgs/deepset-ai/projects/14/views/1)?

Reply:
Done!